tests/authentication/03-check-keys-local.t

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Checks ZMQ keys are created and deleted on shutdown - local.
+. "$(dirname "$0")/test_header"
+set_test_number 10
+init_suite "${TEST_NAME_BASE}" <<'__SUITE_RC__'
+[cylc]
+    cycle point format = %Y
+[scheduling]
+    initial cycle point = 2020
+    [[graph]]
+        R1 = t1
+[runtime]
+    [[t1]]
+        script = sleep 10
+__SUITE_RC__
+
+run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
+
+SRVD="${SUITE_RUN_DIR}/.service"
+
+suite_run_ok "${TEST_NAME_BASE}-run" cylc run "${SUITE_NAME}"
+cylc hold "${SUITE_NAME}"
+
```

Comment:
Sorry @datamel I should have picked this up yesterday, but this local test also needs fixing. The `cylc hold` command only affects waiting tasks, it doesn't stop the suite from finishing and shutting down if there are no more waiting tasks, so this test only works because the auth files are checked during execution of `t1`, and that timing could be flaky under heavy load conditions. 

I'd suggest: start the suite with `cylc run --hold` so that `t1` never runs (then you can delete the `sleep 10` as it will never be used); check the auth files; call `cylc stop` and check auth files again.  (Note the remote test is a bit trickier because a remote task actually has to be submitted to run in order to get the remote auth files installed).

Reply:
Ah grand, thanks for the tips. Sorry this has been a bit onerous to review!

Reply:
Not onerous at all!